src/app/feature/map/search/search.component.ts

```diff
@@ -26,74 +17,59 @@ export class SearchComponent implements OnInit {
   constructor(
     private readonly _mapService: MapService,
     private readonly _mapDataProviderService: MapDataProviderService,
-    private readonly translateService: TranslateService
-  ) {
-    this.destinationGroups = [];
-    this.userLocationGroups = [];
-  }
+    private readonly _translateService: TranslateService
+  ) {}
+
+  public readonly spriteIconsUrl = environment.spriteIconsPath;
+  public readonly labelText = LabelText;
 
-  @Output() set location(value: string) {
+  public setLocation(value: string): void {
     value && value.length && this.findUserLocation(value);
   }
-  @Output() set destination(value: string) {
+  public setDestination(value: string): void {
     value && value.length && this.findDestination(value);
   }
 
-  public featurePoints: GuideMapFeaturePoint[];
-
-  public destinationGroups: IOptionGroup[];
-  public userLocationGroups: IOptionGroup[];
+  public destinationGroups = [] as IOptionGroup[];
+  public userLocationGroups = [] as IOptionGroup[];
 
   public ngOnInit(): void {
-    this._mapDataProviderService.getFeaturePoints().subscribe((data) => {
-      this.featurePoints = data;
-      const qrCodes = data.filter((item) => {
-        return item.properties.category === GuideMapFeaturePointCategory.QrCode;
-      });
-      const rooms = data.filter((item) => {
-        return item.properties.category === GuideMapFeaturePointCategory.Room;
-      });
-      this.userLocationGroups.push({
-        // TODO: group logic
-        groupName: this.translateService.instant('UI.QRCODE_GROUPS.COMMON'),
-        rooms: qrCodes.map(({ properties }) => {
-          return {
-            value: String(properties.id),
-            viewValue: properties.name,
-          };
-        }),
-      });
-      this.destinationGroups.push({
-        groupName: this.translateService.instant('UI.ROOM_GROUPS.CLASSROOMS'),
-        rooms: rooms.map(({ properties }) => {
-          return {
-            value: String(properties.id),
-            viewValue: properties.name,
-          };
-        }),
-      });
+    const { qrCodes$, rooms$ } = this._mapDataProviderService;
+
+    // TODO: split rooms and qr codes into groups (in scope of task: https://trello.com/c/JhfD3q4U/14-app-split-rooms-and-qr-codes-into-groups)
+    qrCodes$.subscribe((qrCodes) => {
```

Comment:
Please unsubscribe here with check UntilDestroy decorator and untilDestroyed(this) operator in pipe()